### PR TITLE
Gripper

### PR DESCRIPTION
**Pull request by AlonSchwierz**
מישהו יודע איפה זה שינקן?

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import edu.wpi.first.wpilibj.Solenoid;
+import frc.robot.Ports;
+import frc.robot.subsystems.LoggedSubsystem;
+
+public class Gripper extends LoggedSubsystem<GripperLoggedInputs> {
+    private final Solenoid leftSolenoid = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Gripper.LEFT_SOLENOID);
+    private final Solenoid rightSolenoid = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Gripper.RIGHT_SOLENOID);
+
+
+    public Gripper(GripperLoggedInputs inputs) {
```
Comment (barel13) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Make this constructor private and add `getInstance`

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    /**
+     * update the logger inputs
+     */
+
+    @Override
+    public void updateInputs() {
+        loggerInputs.leftSolenoidState = leftSolenoid.get();
+
+        loggerInputs.rightSolenoidState = rightSolenoid.get();
```
Comment (barel13) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Why don't use get the values from the methods?

File: src/main/java/frc/robot/subsystems/gripper/GripperLoggedInputs.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package frc.robot.subsystems.gripper;
+
+import org.littletonrobotics.junction.LogTable;
+import org.littletonrobotics.junction.inputs.LoggableInputs;
+
+public class GripperLoggedInputs implements LoggableInputs {
+    boolean leftSolenoidState;
+    boolean rightSolenoidState;
```
Comment (barel13) [inline src/main/java/frc/robot/subsystems/gripper/GripperLoggedInputs.java]: Public

**Review by barel13 — CHANGES_REQUESTED** [review]
I know where is shinken

File: src/main/java/frc/robot/Ports.java [hunk 4]
```diff
@@ -1,5 +1,8 @@
 package frc.robot;
 
 public final class Ports {
-
+public class Gripper{
+    public static final int LEFT_SOLENOID = 0;
+    public static final int RIGHT_SOLENOID = 0;
+}
 }
```
Comment (GaiaZano05) [inline src/main/java/frc/robot/Ports.java:8]: They are on the same side, think of a diffrent name.

**Review by GaiaZano05 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 1 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: done.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 2 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: what do you mean? I am using the methods in order to update the logger inputs.

File: src/main/java/frc/robot/subsystems/gripper/GripperLoggedInputs.java (hunk 3 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/subsystems/gripper/GripperLoggedInputs.java]: done.

File: src/main/java/frc/robot/Ports.java (hunk 4 above)
Comment (AlonSchwierz) [reply src/main/java/frc/robot/Ports.java:8]: done.

**Pull request closed without merging**